core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -185,11 +187,11 @@ class TransactionCoordinator(brokerId: Int,
     } else {
       // caller should have synchronized on txnMetadata already
       txnMetadata.state match {
-        case PrepareAbort | PrepareCommit =>
+        case PrepareAbort | PrepareCommit | Dead =>
```

Comment:
I'm not following what you are suggesting. Yes i think there is a chance of a race in that the thread here gets the metadata from the cache when it is in `pendingState` of `Dead`, but before this thread synchronizes on the `txnMetadata` the `removeFromCacheCallback` completes the  transition to `Dead`